gt/main.c

```diff
@@ -131,36 +129,78 @@ gt_parse_incoming_pkt(struct rte_mbuf *pkt, struct gt_packet_headers *info)
 		info->inner_ip_ver = ETHER_TYPE_IPv4;
 		info->l4_proto = inner_ipv4_hdr->next_proto_id;
 		info->l4_hdr = &inner_ipv4_hdr[1];
+
+		if (rte_ipv4_frag_pkt_is_fragmented(inner_ipv4_hdr)) {
+			info->frag = true;
+			info->l2_len = parsed_len;
+			info->l3_len = sizeof(*inner_ipv4_hdr);
```

Comment:
See comment above -- I'm not sure we can assume that the inner IPv4 header will be here if the packet is fragmented.

Reply:
I think if the packets were fragmented before we encapsulate them, there should an IPv4 header. But, if the packets were fragmented after we encapsulate them, then there wouldn't be such an header, right?